ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2113,3 +2120,108 @@ - (BOOL)isEqual:(id)object {
 }
 
 @end
+
+
+#pragma mark - ORKLocationAnswerFormat
+
+@implementation ORKLocationAnswerFormat
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _useCurrentLocation = YES;
+    }
+    return self;
+}
+
+- (BOOL)isAnswerValidWithString:(nullable NSString *)text {
+    BOOL isValid = [super isAnswerValidWithString:text];
+    
+    MKPlacemark *placemark = [MKPlacemark ork_placemarkWithJSONString:text];
+    
+    if (placemark && CLLocationCoordinate2DIsValid(placemark.coordinate)) {
+        isValid = YES;
+    } else {
+        isValid = NO;
+    }
+    
+    return isValid;
+}
+
+- (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
+    if (![text length]) {
+        return nil;
+    }
+    
+    MKPlacemark *placemark = [MKPlacemark ork_placemarkWithJSONString:text];
+    CLLocationCoordinate2D coordinate = placemark.location.coordinate;
+    NSMutableString *string = [NSMutableString new];
+    
+    if (coordinate.latitude < -90.0) {
+        [string appendString:ORKLocalizedString(@"LOCATION_ALERT_MESSAGE_LATITUDE_BELOW_MINIMUM", nil)];
```

Comment:
>MKPlacemark can hold an invalid coordinate, so I believe this is just some defensive programming. 

I understand that, however, this type error must be a developing error; Just using the UI, a user generally cannot create an invalid `MKPlacemark`. Even if MapKit sends out an invalid one, which is a *system error*, nothing todo with the user. It is inappropriate to display error message to user like this: user simply cannot resolve it. 

If an invalid `MKPlacemark` is not from MapKit, it is generated by code, maybe we can warn developer in an early stage.  

How about creating `ORKPlaceMark` as the subclass of `MKPlacemark`, and override the init method to prevent creating a placemark with invalid `coordinates`. What do you think?